frontend/ui/widget/inputtext.lua

```diff
@@ -634,6 +639,33 @@ function InputText:onKeyPress(key)
     else
         handled = false
     end
+    if not handled and (key["ScreenKB"] or key["Shift"]) then
+        handled = true
+        if key["Back"] then
+            self:delChar()
+        elseif key["Left"] then
+            self:leftChar()
```

Comment:
So, in effect, this makes all these keys work as-is when you are still holding <kbd>Shift</kbd> pressed, which I guess is practical and fine by me.
Except for Home and End - but ok. But we're hacking Shift+Home to hide/show keyboard, so preventing further implementations of something balanced. Just mentionning that.

About Shift + <kbd>.</kbd> outputing a comma, I checked that (with another key, as my keyboard does not have the dot available as a plain key) that it seems there is no impact, and we would never get that comma with a real keyboard on the emulator. Not sure why :) and how <kbd>Shift</kbd> is handled.
I checked with the virtucal keyboard, by putting <kbd>.</kbd> in both the first panel and the panel we get when we press <kbd>Shift</kbd>, and I keep getting <kbd>.</kbd>. Good. (But of course, it is wrapped bt hasSymKeys().... realized that too late).

Reply:
> that it seems there is no impact, and we would never get that comma with a real keyboard on the emulator. Not sure why :)

SDL gives us a full string so we don't have to worry about any of the complexities involved in a keypress not mapping directly to a character. (like for typing é)

Reply:
So the TL;DR here is, all good, right?

> as my keyboard does not have the dot available as a plain key

wild. Buy a better keyboard ;)

btw, tested now, working as expected.